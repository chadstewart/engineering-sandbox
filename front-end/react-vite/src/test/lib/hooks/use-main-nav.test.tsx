/**
 * @vitest-environment happy-dom
 */

import useMainNav from "@/lib/hooks/use-main-nav";
import { renderHook } from "@testing-library/react";
import { describe, expect, it } from "vitest";

describe("Utility Function Hook: UseMainNav", () => {
  it("Matches the snapshot of useMainNav", () => {
    const {
      result: { current: variableToSnapshot }
    } = renderHook(useMainNav);

    expect(variableToSnapshot).toMatchSnapshot();
  });
});
