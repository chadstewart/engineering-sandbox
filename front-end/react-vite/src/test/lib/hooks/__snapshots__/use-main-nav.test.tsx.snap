// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Utility Function Hook: UseMainNav > Matches the snapshot of useMainNav 1`] = `
[
  {
    "icon": [Function],
    "iconAlt": "Home page link icon",
    "text": "Home",
    "url": "/",
  },
  {
    "icon": [Function],
    "iconAlt": "Cat page link icon",
    "text": "Cute Animals",
    "url": "/cute-animals",
  },
  {
    "icon": [Function],
    "iconAlt": "Order dropdown link icon",
    "params": {
      "page": 1,
    },
    "text": "Orders",
    "url": "/orders/details/$page",
  },
  {
    "icon": [Function],
    "iconAlt": "Product page link icon",
    "params": {
      "page": 1,
    },
    "text": "Products",
    "url": "/products/details/$page",
  },
  {
    "children": [
      {
        "icon": [Function],
        "iconAlt": "Details page link icon",
        "params": {
          "page": 1,
        },
        "text": "Details",
        "url": "/customers/details/$page",
      },
      {
        "icon": [Function],
        "iconAlt": "Add Customer link icon",
        "isProtectedLink": true,
        "text": "Add Customer",
        "url": "/customers/add-customer",
      },
    ],
    "icon": [Function],
    "iconAlt": "Customer page link icon",
    "text": "Customers",
  },
  {
    "icon": [Function],
    "iconAlt": "About page link icon",
    "text": "About",
    "url": "/about",
  },
  {
    "icon": [Function],
    "iconAlt": "Architecture page link icon",
    "text": "Architecture",
    "url": "/architecture",
  },
]
`;
